src/transformer/remove_comments.cpp

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+#include "transformer/remove_comments.hpp"
+
+namespace ddwaf::transformer {
+
+bool remove_comments::transform_impl(cow_string &str)
+{
+    enum class comment_type { unknown, html, c, eol } type = comment_type::unknown;
+
+    bool comment_found = false;
+
+    std::size_t read = 0;
+    std::size_t write = 0;
+    while (read < str.length()) {
+        type = comment_type::unknown;
+        while (read < str.length()) {
+            if (str.at(read) == '<' && read + 3 < str.length() && str.at(read + 1) == '!' &&
```

Comment:
Since all strings on this and on every other transformers are `cow_string` would it be possible to have a method to avoid having to check things like `read + x < lengh && read + x == Y`.